tests/portal-chooser/recon-session.sh

```
@@ -0,0 +1,325 @@
+
+Xvfb ":$DISPLAY_NUM" -screen 0 "${RECON_SCREEN:-1920x1200}x24" -nolisten tcp >"$OUT/xvfb.log" 2>&1 &
+XVFB_PID=$!
+export DISPLAY=":$DISPLAY_NUM"
+
+cleanup() {
+  [ -n "${DBUS_PID:-}" ] && kill -TERM "$DBUS_PID" 2>/dev/null
+  [ -n "${APP_PID:-}" ] && { kill -TERM -- "-$APP_PID" 2>/dev/null; sleep 1; kill -KILL -- "-$APP_PID" 2>/dev/null; }
+  [ -n "${PORTAL_PID:-}" ] && kill -INT "$PORTAL_PID" 2>/dev/null
+  [ -n "${XVFB_PID:-}" ] && kill -TERM "$XVFB_PID" 2>/dev/null
+  wait 2>/dev/null
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)